python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -0,0 +1,335 @@
+                if cidr is not compare:
+                    if cidr.overlaps(compare):
+                        addresses.append([ERROR_KEY, key, f"{cidr} is not valid as it overlaps with {compare}"])
+
+        hook.end()
+
+        return addresses
+
+    return _validate_cidr_ranges
+
+
+def dyn_validate_resource_permissions(service_principle_ids, resources):
+    def _validate_resource_permissions(cmd,
+                                       namespace):  # pylint: disable=unused-argument
+        errors = []
+
+        hook = cmd.cli_ctx.get_progress_controller()
+        hook.add(message="Validating resource permissions")
+
+        for sp_id in service_principle_ids:
+            for role in sorted(resources):
+                for resource in resources[role]:
+                    try:
+                        resource_contributor_exists = has_role_assignment_on_resource(cmd.cli_ctx,
+                                                                                      resource,
+                                                                                      sp_id,
+                                                                                      role)
+                        if not resource_contributor_exists:
+                            parts = parse_resource_id(resource)
+                            errors.append(["Resource Permissions",
+                                           parts['type'],
+                                           f"Resource {parts['name']} is missing role assignment " +
+                                           f"{role} for service principal {sp_id} " +
+                                           "(These roles will be automatically added during cluster creation)"])
+                    except CloudError as e:
+                        logger.error(e.message)
+                        raise
+        hook.end()
+        return errors
+    return _validate_resource_permissions
+
+
+def dyn_validate_version():
+    def _validate_version(cmd,
+                          namespace):  # pylint: disable=unused-argument
+        errors = []
+
+        hook = cmd.cli_ctx.get_progress_controller()
+        hook.add(message="Validating OpenShift Version")
+
+        if namespace.location is None:
+            get_default_location_from_resource_group(cmd, namespace)
+
+        versions = azext_aro.custom.aro_get_versions(namespace.client, namespace.location)
+
+        found = False
+        for version in versions:
+            if version == namespace.version:
+                found = True
+                break
+
+        if not found:
+            errors.append(["OpenShift Version",
+                           namespace.version,
+                           f"{namespace.version} is not a valid version, valid versions are {versions}"])
+
+        hook.end()
+        return errors
+    return _validate_version
+
+
+def validate_cluster_create(cmd,  # pylint: disable=unused-argument
```

Comment:
Unused arguments should be able to be turned off by prefixing them with an underscore.